indra/newview/lllocalbitmaps.cpp

```diff
@@ -1137,6 +1138,20 @@ void LLLocalBitmapMgr::delUnit(LLUUID tracking_id)
 	}
 }
 
+LLUUID LLLocalBitmapMgr::getTrackingID(const LLUUID& world_id) const
```

Comment:
Suggestion: implement `class LLViewerLocalTexture: public LLViewerFetchedTexture` that stores tracking id and replace `LLViewerFetchedTexture` in `LLLocalBitmap::updateSelf`
```
                    LLPointer<LLViewerFetchedTexture> texture = new LLViewerFetchedTexture
                        ("file://"+mFilename, FTT_LOCAL_FILE, mWorldID, LL_LOCAL_USE_MIPMAPS);
```
This way there won't be a need to search, just cast to LLViewerLocalTexture and get the tracking id.

Reply:
I think I'd rather eat the search than add a sketchy typecast.  Once all this is functional it's going to get some heavy profiling over the summer and then we'll find out if the search is really relevant performance wise.